services/src/unit-test/java/uk/gov/moj/sdt/services/BulkSubmissionServiceTest.java

```diff
@@ -367,4 +456,108 @@ private void addValidIndividualRequest(final IBulkSubmission bulkSubmission, fin
+
+
+        bulkSubmission.setErrorMessagesCache(errorMessagesCache);
+
+        Field errorMessagesCacheField = BulkSubmissionService.class.getDeclaredField("errorMessagesCache");
+        errorMessagesCacheField.setAccessible(true);
+        ICacheable result = (ICacheable) errorMessagesCacheField.get(bulkSubmission);
+
+        assertEquals(errorMessagesCache, result, "The errorMessagesCache should be set correctly");
+    }
+
+
+    @Test
+    public void setMockSdtBulkReferenceGeneratorTest() throws Exception {
+
+        bulkSubmissionService.setSdtBulkReferenceGenerator(mockSdtBulkReferenceGenerator);
+
+        Object result = this.getAccessibleField(BulkSubmissionService.class,"sdtBulkReferenceGenerator",
+                                                SdtBulkReferenceGenerator.class,bulkSubmissionService);
+
+        assertEquals(mockSdtBulkReferenceGenerator, result, "The sdtBulkReferenceGenerator should be set correctly");
+    }
+
+    @Test
+    public void setTargetApplicationDaoTest(){
+
+        bulkSubmissionService.setTargetApplicationDao(mockTargetApplicationDao);
+
+        ITargetApplicationDao result = bulkSubmissionService.getTargetApplicationDao();
+
+        assertEquals(mockTargetApplicationDao, result, "The TargetApplication should be set correctly");
+    }
+
+    @Test
+    public void setIndividualRequestXmlParserTest(){
+
+        IndividualRequestsXmlParser individualRequestsXmlParserMock = mock(IndividualRequestsXmlParser.class);
+        bulkSubmissionService.setIndividualRequestsXmlparser(individualRequestsXmlParserMock);
+
+        IndividualRequestsXmlParser result = bulkSubmissionService.getIndividualRequestsXmlParser();
+
+        assertEquals(individualRequestsXmlParserMock, result, "The TargetApplication should be set correctly");
+    }
+
+    @Test
+    public void setMessagingUtilityTest(){
+
+        bulkSubmissionService.setMessagingUtility(mockMessagingUtility);
+
+        IMessagingUtility result = bulkSubmissionService.getMessagingUtility();
+
+        assertEquals(mockMessagingUtility, result, "The MessageUtility should be set correctly");
+    }
+
+    @Test
+    public void setGenericDaoTest(){
+
+        bulkSubmissionService.setGenericDao(mockGenericDao);
+
+        IGenericDao result = bulkSubmissionService.getGenericDao();
+
+        assertEquals(mockGenericDao, result, "The GenericDao should be set correctly");
+    }
+
+    @Test
+    public void setBulkCustomerDaoTest(){
+
+        bulkSubmissionService.setBulkCustomerDao(mockBulkCustomerDao);
+
+        IBulkCustomerDao result = bulkSubmissionService.getBulkCustomerDao();
+
+        assertEquals(mockBulkCustomerDao, result, "The GenericDao should be set correctly");
+    }
+
+
+    /**
+     * Method to check logging message in log list
+     * @param logList
+     * @param message
+     * @return boolean
+     */
+    private static boolean verifyLog(List<ILoggingEvent> logList, String message) {
```

Comment:
This method will always iterate through all the messages in the log before returning.  Once the message has been found there's no need to check the others, so add something to terminate the loop once it's been found.

Reply:
Added a break;